Sources/TuistSigning/SigningCipher.swift

```diff
@@ -0,0 +1,134 @@
+import Basic
+import CryptoSwift
+import Foundation
+import TuistCore
+import TuistSupport
+
+enum SigningCipherError: FatalError, Equatable {
+    case failedToEncrypt
+    case masterKeyNotFound(AbsolutePath)
+    case rootDirectoryNotFound(AbsolutePath)
+
+    var type: ErrorType { .abort }
+
+    var description: String {
+        switch self {
+        case .failedToEncrypt:
+            return "Encryption failed"
+        case let .masterKeyNotFound(masterKeyPath):
+            return "Could not find master.key at \(masterKeyPath.pathString)"
+        case let .rootDirectoryNotFound(fromPath):
+            return "Could not find root directory from \(fromPath.pathString)"
+        }
+    }
+}
+
+/// SigningCiphering handles all encryption/decryption of files needed for signing (certificates, profiles, etc.)
+public protocol SigningCiphering {
+    /// Encrypts all signing files at `Tuist/Signing`
+    func encryptSigning(at path: AbsolutePath) throws
+    /// Decrypts all signing files at `Tuist/Signing`
+    func decryptSigning(at path: AbsolutePath) throws
+}
+
+public final class SigningCipher: SigningCiphering {
+    private let rootDirectoryLocator: RootDirectoryLocating
+
+    /// Public initializer
+    public init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator()) {
+        self.rootDirectoryLocator = rootDirectoryLocator
+    }
+
+    public func encryptSigning(at path: AbsolutePath) throws {
+        let (signingKeyFiles, masterKey) = try signingData(at: path)
+        let cipheredKeys = try signingKeyFiles
+            .map(FileHandler.shared.readFile)
+            .map { try encryptData($0, masterKey: masterKey) }
+
+        try zip(cipheredKeys, signingKeyFiles).forEach {
+            try $0.write(to: $1.url)
+        }
+    }
+
+    public func decryptSigning(at path: AbsolutePath) throws {
+        let (signingKeyFiles, masterKey) = try signingData(at: path)
+        let decipheredKeys = try signingKeyFiles
+            .map(FileHandler.shared.readFile)
+            .map {
+                try decryptData($0, masterKey: masterKey)
+            }
+
+        try zip(decipheredKeys, signingKeyFiles).forEach {
+            try $0.write(to: $1.url)
```

Comment:
It might be a good idea to introduce some logging so that the user gets some feedback as the work for encrypt/decrypt is going on. You can introduce `debug` messages for more detailed logs which only get printed when `--verbose` is passed to any of the commands - this will help users of Tuist to debug what is wrong.

Reply:
You're definitely right, I have not got used to using it just yet, but I definitely should since it might be crucial for us and for the end user!